### PR TITLE
Fix migrator

### PR DESCRIPTION
**Pull request by frankiebee**
as i was writing my own migration I noticed that the migration would run but the localStorage would not update so i fixed that here 
the issue was on line:19
`.then(() => versionedData)`
this seemed to be returning the original versionedData that was supplied as an argument to the function instead of the migratedData. I also swapped out  asyncQ.eachSeries for asyncQ.waterfall so that results of previous migrations if any are used in the next migration that is run.

### CODE REVIEW

**Comment by kumavis** [comment]
Thats a pretty serious bug. Can you add a test that fails without this change?

**Comment by frankiebee** [comment]
yup I'll do that now

File: app/scripts/lib/migrator/index.js [hunk 1]
```diff
@@ -13,10 +13,16 @@ class Migrator {
...
+    const migrations = remaining.map((migration, i) => {
+      if (i === 0) return this.runMigration.bind(this, migration, versionedData)
+      return this.runMigration.bind(this, migration)
+    })
 
     return (
-      asyncQ.eachSeries(remaining, (migration) => this.runMigration(versionedData, migration))
-      .then(() => versionedData)
+      asyncQ.waterfall(migrations)
+      .then((migratedData) => Promise.resolve(migratedData))
```
Comment (kumavis) [inline app/scripts/lib/migrator/index.js]: `.then((migratedData) => Promise.resolve(migratedData))`
this can be removed - doesnt do anything

**Review by kumavis — CHANGES_REQUESTED** [review]
good find

the code is rather tricky to follow

im sure we can find a more straightforward and readable way to run the migrations

**Comment by kumavis** [comment]
yeah thats about as clean as it gets w/o es7. went with es7!  rejoice! now we can use that where ever we want

**Pull request closed without merging**